### PR TITLE
perf(oauth): add question model for oauth

### PR DESCRIPTION
**Pull request by KennethBWSong**

### CODE REVIEW

File: packages/fx-core/src/question/other.ts [hunk 1]
```diff
@@ -994,3 +994,58 @@ export function apiSpecApiKeyQuestion(): IQTreeNode {
...
+export function OauthClientSecretConfirmQestion(): ConfirmQuestion {
+  return {
+    name: QuestionNames.OauthClientSecretConfirm,
+    title: getLocalizedString("core.createProjectQuestion.OauthClientSecretConfirm"),
+    type: "confirm",
+    default: true,
+  };
+}
+
+export function oauthClientSecretQuestion(): IQTreeNode {
```
Comment (SLdragon) [inline packages/fx-core/src/question/other.ts:1007]: Maybe we should also ask user for clientId?

File: packages/fx-core/src/question/other.ts [hunk 2]
```diff
@@ -994,3 +994,58 @@ export function apiSpecApiKeyQuestion(): IQTreeNode {
...
+      cliDescription: "Oauth client secret for OpenAPI spec.",
+      forgetLastValue: true,
+      validation: {
+        validFunc: (input: string): string | undefined => {
+          const pattern = /^(\w){10,128}/g;
+          const match = pattern.test(input);
+
+          const result = match
+            ? undefined
+            : getLocalizedString("core.createProjectQuestion.invalidApiKey.message");
```
Comment (SLdragon) [inline packages/fx-core/src/question/other.ts:1023]: rename invalidApiKey.message to invalidSecret?

Comment (KennethBWSong) [reply packages/fx-core/src/question/other.ts]: Kept here since the message is the same. Will update if we want to have difference message.

**Review by SLdragon — APPROVED** [review]

**Review by jayzhang — DISMISSED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -363,6 +363,9 @@
   "core.createProjectQuestion.OpenAIPluginDomain.placeholder": "Enter your website domain or manifest URL",
   "core.createProjectQuestion.ApiKey": "Enter API Key in OpenAPI Description Document",
   "core.createProjectQuestion.ApiKeyConfirm": "Teams Toolkit will upload the API key to Teams Developer Portal. The API key will be used by Teams client to securely access your API in runtime when API-Based Message Extension is triggered. Teams Toolkit will not store your API key.",
+  "core.createProjectQuestion.OauthClientId": "Enter client id for Oauth registration in OpenAPI Description Document",
+  "core.createProjectQuestion.OauthClientSecret": "Enter client secret for Oauth registration in OpenAPI Description Document",
+  "core.createProjectQuestion.OauthClientSecretConfirm": "Teams Toolkit will upload the client id/secret for Oauth Registration to Teams Developer Portal. The client id/secret will be used by Teams client to securely access your API in runtime when API-Based Message Extension is triggered. Teams Toolkit will not store your client id/secret.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:368]: Revised:
Teams Toolkit uploads the client id/secret for Oauth Registration to Teams Developer Portal. It is used by Teams client to securely access your API at runtime when API-Based Message Extension is triggered. Teams Toolkit doesn't store your client id/secret.

Comment (KennethBWSong) [reply packages/fx-core/resource/package.nls.json]: Updated, thanks!

**Review by supkasar — APPROVED** [review]

**Review by SLdragon — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Pull request merged**